km61/Mehediuk_Kateryna/laboratory1.sql

```diff
@@ -1,2 +1,132 @@
 -- LABORATORY WORK 1
 -- BY Mehediuk_Kateryna
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+оновлювати дані в таблицях
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER TO Kate
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP"
+QVOTAS
+ALTER USER Kate  QVOTA 100M IN USERS
+ROLES
+GRANT "CONNECT" TO Kate
+SYSTEM PRIVILEGES
+GRANT CREATE TO UPDATE TO Kate
+
+
+
+      
+      
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина дивиться кіно.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Скільком покупцям продано найдешевший товар?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Яка країна, у якій живуть постачальники має найдовшу назву?
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
```

Comment:
Select vend_country
From vendors
Where length(trim(vend_country)) in (
Select max(length(trim(vend_country)))
From vendors );